### ISSUE TITLE
BulkInsert messes primary keys

### ISSUE THREAD

**Issue by pako1337** [borisdj/EFCore.BulkExtensions#242]
hey,
In general: I have list of entities, where each entity has sub-list of entities. So, like documented, i try to do two bulk inserts:

`    await database.BulkInserAsync(parents, new BulkConfig { PreserveInsertOrder = true, SetOutputIdentity = true});`

and then I process identities and try to insert children:

```
foreach (var parent in parents)
{
    foreach (var child in parent.Children)
    {
        child.ParentId = parent.Id;
    }
}

await database.BulkInsertAsync(parents.SelectMany(s => s.Children).ToList());
```

Unfortunately, it seems to mess up Ids - parents entry say for example that its id is 123, but in database it is actualy 125, and score with id 125 in code has id 123 in database etc. Which completely messes with saving data - children get persisted for wrong parents!

Is that a known issue, am i doing something wrong with `PreserveInsertOrder` and `SetOutputIdentity`?

**Comment by pako1337** [comment]
Hmm, seems like it is reversing Ids order? I can see, quite consistently, that first parent's children get last parent's id, second parent's children get second-to-last parent's id etc.

**Comment by pako1337** [comment]
Ah, never mind! It finally clicked what I was missing from the documentation. It is not just the order that matters, but I *have to*, before saving, enter some fake Ids for parents that would indicate the order in which they were added to the list. The order in the list itself did not matter, algorithm picked them from the end of the list it seems. But it was in the documentation, I just didn't get it until I found an error and figured out what exactly is going on until documentation text clicked in my brain.

Thanks for the library,
Jarek
